src/askui/tools/askui/askui_controller.py

```diff
@@ -724,3 +725,487 @@ def run_command(self, command: str, timeout_ms: int = 30000) -> None:
+        """
+        Get a list of running processes.
+
+        Args:
+            get_extended_info (bool, optional): Whether to include
+                extended process information.
+                Defaults to False.
+
+        Returns:
+            controller_v1_pbs.Response_GetProcessList: Process list response containing:
+                - processes: List of ProcessInfo objects
+
+        Raises:
+            AssertionError: If the gRPC stub is not initialized.
+        """
+        assert isinstance(self._stub, controller_v1.ControllerAPIStub), (
+            "Stub is not initialized"
+        )
+
+        self._reporter.add_message("AgentOS", f"get_process_list({get_extended_info})")
+
+        response: controller_v1_pbs.Response_GetProcessList = self._stub.GetProcessList(
+            controller_v1_pbs.Request_GetProcessList(getExtendedInfo=get_extended_info)
+        )
+
+        return response
+
+    @telemetry.record_call()
+    def get_window_list(
+        self, process_id: int
+    ) -> controller_v1_pbs.Response_GetWindowList:
+        """
+        Get a list of windows for a specific process.
+
+        Args:
+            process_id (int): The ID of the process to get windows for.
+
+        Returns:
+            controller_v1_pbs.Response_GetWindowList: Window list response containing:
+                - windows: List of WindowInfo objects with ID and name
+
+        Raises:
+            AssertionError: If the gRPC stub is not initialized.
+        """
+        assert isinstance(self._stub, controller_v1.ControllerAPIStub), (
+            "Stub is not initialized"
+        )
+
+        self._reporter.add_message("AgentOS", f"get_window_list({process_id})")
+
+        response: controller_v1_pbs.Response_GetWindowList = self._stub.GetWindowList(
+            controller_v1_pbs.Request_GetWindowList(processID=process_id)
+        )
+
+        return response
+
+    @telemetry.record_call()
+    def get_automation_target_list(
+        self,
+    ) -> controller_v1_pbs.Response_GetAutomationTargetList:
+        """
+        Get a list of available automation targets.
+
+        Returns:
+            controller_v1_pbs.Response_GetAutomationTargetList:
+                Automation target list response:
+                - targets: List of AutomationTarget objects
+
+        Raises:
+            AssertionError: If the gRPC stub is not initialized.
+        """
+        assert isinstance(self._stub, controller_v1.ControllerAPIStub), (
+            "Stub is not initialized"
+        )
+
+        self._reporter.add_message("AgentOS", "get_automation_target_list()")
+
+        response: controller_v1_pbs.Response_GetAutomationTargetList = (
+            self._stub.GetAutomationTargetList(controller_v1_pbs.Request_Void())
+        )
+
+        return response
+
+    @telemetry.record_call()
+    def set_test_configuration(
+        self,
+        default_capture_parameters: controller_v1_pbs.CaptureParameters | None = None,

```

Comment:
Did you test if this works with `telemetry.record_call()`? I have this question for all methods where we pass more complex parameters here as they all need to be serializable.